nmostesting/suites/IS0502Test.py

```diff
@@ -952,6 +952,81 @@ def test_16(self, test):
 
         return test.PASS()
 
+    def test_17(self, test):
+        """IS-05 transportfile ts-refclk matches IS-04 Source and Node"""
+
+        for resource_type in ["senders", "flows", "sources"]:
+            valid, result = self.get_is04_resources(resource_type)
+            if not valid:
+                return test.FAIL(result)
+
+        valid, result = self.get_is05_resources("senders")
+        if not valid:
+            return test.FAIL(result)
+
+        if len(self.is04_resources["senders"]) == 0:
+            return test.UNCLEAR("Could not find any IS-05 Senders to test")
+
+        flow_map = {flow["id"]: flow for flow in self.is04_resources["flows"]}
+        source_map = {source["id"]: source for source in self.is04_resources["sources"]}
+
+        valid, resource = self.do_request("GET", self.node_url + "self")
+        if not valid:
+            return test.FAIL("Node API did not respond as expected: {}".format(resource))
+
+        try:
+            node_self = resource.json()
+        except json.JSONDecodeError:
+            return test.FAIL("Non-JSON response returned from Node API")
+
+        clock_map = {clock["name"]: clock for clock in node_self["clocks"]}
+
+        for resource in self.is04_resources["senders"]:
+            if not resource["transport"].startswith("urn:x-nmos:transport:rtp"):
+                continue
+            if resource["flow_id"] is None:
+                continue
+
+            flow = flow_map[resource["flow_id"]]
+            source = source_map[flow["source_id"]]
+
+            is05_transport_file = self.is05_resources["transport_files"][resource["id"]]
+            if is05_transport_file is None:
+                return test.FAIL("Unable to download transportfile for Sender {}".format(resource["id"]))
+
+            found_refclk = False
+            for sdp_line in is05_transport_file.split("\n"):
+                sdp_line = sdp_line.replace("\r", "")
+                ts_refclk = re.search(r"^a=ts-refclk:(.+)$", sdp_line)
+                if not ts_refclk:
+                    continue
+                found_refclk = True
+                if source["clock_name"] is None:
+                    return test.FAIL("SDP file includes ts-refclk but Source {} does not indicate a clock_name"
```

Comment:
By the way, is having a non-`null` clock a requirement of a "periodic Source", like having a grain rate?

Reply:
I guess it depends. If the signal originated at the Sender then it probably ought to signal a clock (even if internal), but if the signal originated elsewhere and was just transformed by the Sender, the Sender's Node may not lock the signal to its own clock first.